### PR TITLE
Use SpiceDB directly instead of NATS KV for ZedToken updates

### PR DESCRIPTION
**Pull request by jnschaeffer**
NATS introduces an additional dependency for getting ZedToken updates that we don't really need if we can just use the SpiceDB Watch API. This PR updates the permissions-api server to use SpiceDB Watch instead for ZedToken updates.

### CODE REVIEW

File: internal/query/zedtokens.go [hunk 1]
```diff
@@ -19,48 +15,49 @@ const (
...
+			relationship.Subject.Object.ObjectId,
+			zedToken,
+		)
 	}
+}
 
-	ttl := status.TTL()
+// initZedTokenCache creates a new LRU cache that watches SpiceDB for ZedToken updates.
+func (e *engine) initZedTokenCache(ctx context.Context) error {
+	ttl := time.Minute
```
Comment (jnschaeffer) [inline internal/query/zedtokens.go:38]: Note to self: we should make this configurable based on the revision quantization interval.

File: internal/query/zedtokens.go [hunk 2]
```diff
@@ -19,48 +15,49 @@ const (
...
-			if entry == nil {
-				continue
-			}
-
-			key := entry.Key()
-			value := string(entry.Value())
+		for {
+			resp, err := watchClient.Recv()
+			if err != nil {
+				e.logger.Errorf("error receiving updates", "error", err)
```
Comment (mikemrm) [inline internal/query/zedtokens.go:53]: should we exit this goroutine if the error is a context cancelled?

**Review by mikemrm — APPROVED** [review]
looks good

**Review by bailinhe — COMMENTED** [review]
looks like one of the tests need to be updated

https://github.com/infratographer/permissions-api/actions/runs/9097856914/job/25006681107?pr=256#step:3:2763

**Comment by jnschaeffer** [comment]
Closing in favor of #257.

**Pull request closed without merging**